src/main/java/subway/line/entity/Sections.java

```diff
@@ -0,0 +1,64 @@
+package subway.line.entity;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import javax.persistence.OrderBy;
+import java.util.ArrayList;
+import java.util.List;
+
+@Embeddable
+public class Sections {
+
+    @OrderBy("id ASC")
+    @OneToMany(mappedBy = "line", cascade = CascadeType.ALL)
+    private List<Section> sections = new ArrayList<>();;
+
+    protected Sections() {
+    }
+
+    private Sections(final List<Section> sections) {
+        this.sections = sections;
+    }
+
+    public void addSection(final Section section) {
+        this.sections.add(section);
+    }
+
+    public void removeSection(final Section section) {
+        this.sections.remove(section);
+    }
+
+    public static Sections from(final List<Section> sections) {
+        return new Sections(sections);
+    }
+
+    public int size() {
+        return sections.size();
+    }
+
+    public boolean isLastStation(final long stationId) {
+        if (sections.isEmpty()) return false;
+
+        Section lastSection = sections.get(sections.size() - 1);
+        return lastSection.getDownStation().getId().equals(stationId);
+    }
+
+    public boolean isSectionRegistered(final long upStationId, final long downStationId) {
+        return this.getSections().stream()
+                .anyMatch(section ->
+                        section.getUpStation().getId().equals(upStationId) && section.getDownStation().getId().equals(downStationId)
+                );
+    }
+
+    public void ensureNoDuplicateDownStation(final Section newSection) {
+        final boolean hasDuplicateDownStation = this.getSections().stream()
+                .anyMatch(section -> newSection.getDownStation().getId().equals(section.getUpStation().getId()));
```

Comment:
>객체지향 생활 체조 
규칙 4: 한 줄에 점을 하나만 찍는다.

위 규칙을 디미터 법칙을 적용하라는 규칙인데요
getter를 활용하기 보다 객체에게 메시지를 보내면 어떨까요?
```suggestion
                .anyMatch(section -> newSection.isSameDownStation(section.getUpStation()));
```

Reply:
아하 안그래도 저 부분을 작성하면서 메서드로 빼면 좋을지 고민했었어요
법칙 기억하도록 하겠습니다 😄